File: 1796_SecondLargest/solution.java
class Solution {
    public int secondHighest(String s) {
        int largest = -1;
        int secondLargest = -1;
        for(char c : s.toCharArray()){
            if(Character.isDigit(c)){
                int digit = c -'0';
                if(digit > largest){
                    secondLargest = largest;
                    largest = digit;
                }
                else if(digit > secondLargest && digit < largest){
                    secondLargest = digit;
                }
            }
        }
        return secondLargest;
    }
}

File: 1796_SecondLargest/solution.py
class Solution:
    def secondHighest(self, s: str) -> int:
        largest = -1
        secondLargest = -1

        for c in s:
            if c.isdigit():
                digit = int(c)
                if digit > largest:
                    secondLargest = largest
                    largest = digit
                elif largest > digit > secondLargest:
                    secondLargest = digit

        return secondLargest